tests/container-copy

```diff
@@ -0,0 +1,70 @@
+#!/bin/sh
+set -eu
+
+# Install and setup LXD.
+install_lxd
+lxd init --auto
+
+# Create a loop device that we can mount as a source for a dir storage pool.
+test_dir="$(mktemp -d XXX.test)"
+loopimg="$(mktemp "${test_dir}/XXX.img")"
+truncate -s 5G "${loopimg}"
+loopdev="$(losetup --show --find "${loopimg}")"
+
+# Begin Test.
+set -x
+
+# Launch a VM which will be the target of the copy.
+lxc launch "${TEST_IMG:-ubuntu-minimal-daily:22.04}" target --vm
+waitInstanceBooted target
+
+# Add the target VM as a remote LXD.
+target_addr="$(lxc ls -f csv -c 4 target | grep -v lxdbr | cut -d' ' -f1 | sed 's/"//g')"
+lxc exec target -- snap install lxd --channel "${LXD_SNAP_CHANNEL}"
+lxc exec target -- lxd init --auto
+lxc exec target -- lxc config set core.https_address=:8443
+
+# Use `lxc remote add` to generate client certs, in case they don't exist.
+! lxc remote add generate-certs not-a-url --accept-certificate 2>/dev/null || true
+
+lxc file push ~/snap/lxd/common/config/client.crt target/root/
+lxc exec target -- lxc config trust add test /root/client.crt
+lxc remote add target "https://${target_addr}:8443" --accept-certificate
+
+# Run the test for various filesystems.
+for fs in "xfs" "btrfs" "ext4" ; do
+  mkfs.${fs} "${loopdev}"
+  disk_dir="$(mktemp -d "${test_dir}/XXX.disk")"
+  mount "${loopdev}" "${disk_dir}"
+  rm -rf "${disk_dir}/lost+found"
+
+  # Create a storage pool with the mounted disk.
+  lxc storage create "${fs}" dir source="${PWD}/${disk_dir}"
+
+  # Create a container which we will copy to another target.
+  lxc launch "${TEST_IMG:-ubuntu-minimal-daily:22.04}" c1 --storage "${fs}"
+
+  # Copy the container to the remote LXD.
+  lxc copy c1 target:c1 --stateless --refresh --mode=relay
+
+  # Ensure the container exists.
+  lxc config show target:c1 > /dev/null
+
+  # Clean up the storage pool for this run.
+  lxc delete c1 -f
+  lxc delete target:c1 -f
+  lxc storage delete "${fs}"
+
+  # Reset the loop device.
+  umount "${loopdev}"
+  wipefs -a "${loopdev}"
+done
+
+# Clean up LXD and remove the loop device.
```

Comment:
@simondeziel should this be a cleanup function?

Reply:
@tomponline tests defining an `extra_cleanup()` function will have their cleanup done only on successful runs which is fine but they will run that cleanup with `set +e` meaning if a cleanup step fails, it won't fail the test.

I'm now questioning my own decision of doing this as being able to cleanup properly feels important too.